src/spatial.jl

```diff
@@ -27,6 +27,11 @@ Setting one or more values to `Inf` gives no boundary in that dimension.
 """
 struct CubicBoundary{T}
     side_lengths::SVector{3, T}
+    CubicBoundary(side_lengths::SVector{3, E}) where E = (
+        all(a -> a >= 0*unit(a),  side_lengths) ?
+        new{E}(side_lengths) :
+        throw(DomainError("Side lengths need to be positive, got $side_lengths"))
```

Comment:
Maybe change to "Side lengths can't be negative" to match the check.

Reply:
My bad, forgot to change this to include the zero case.